pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -160,8 +160,8 @@ func (s *Server) initK8SConfigStore(args *PilotArgs) error {
 		if s.statusManager == nil && features.EnableGatewayAPIStatus {
 			s.initStatusManager(args)
 		}
-		gwc := gateway.NewController(s.kubeClient, configController, s.kubeClient.CrdWatcher().WaitForCRD,
-			s.environment.CredentialsController, args.RegistryOptions.KubeOptions)
+		args.RegistryOptions.KubeOptions.KrtDebugger = args.KrtDebugger
+		gwc := gateway.NewController(s.kubeClient, s.kubeClient.CrdWatcher().WaitForCRD, args.RegistryOptions.KubeOptions, s.XDSServer)
 		s.environment.GatewayAPIController = gwc
```

Comment:
@howardjohn I was wondering, `initK8SConfigStore` is called for every kubernetes config source which could be multiple, but here we're storing (and overwriting) a single general `GatewayAPIController`, so we only every keep the last `GatewayAPIController` reference, this means that we only every `Reconcile` this last one, probably introducing bugs whenever working with multiple kuberentes config sources._

I don't know if this is tested or even supported at all.

Reply:
👍 I can open a PR getting rid of it